sdks/go/pkg/beam/io/parquetio/parquetio.go

```diff
@@ -0,0 +1,140 @@
+package parquetio
+
+import (
+	"context"
+	"io/ioutil"
+	"reflect"
+	"strings"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/io/filesystem"
+	"github.com/xitongsys/parquet-go-source/buffer"
+	"github.com/xitongsys/parquet-go/reader"
+	"github.com/xitongsys/parquet-go/writer"
+)
+
+func init() {
+	beam.RegisterFunction(expandFn)
+	beam.RegisterType(reflect.TypeOf((*parquetReadFn)(nil)).Elem())
+	beam.RegisterType(reflect.TypeOf((*parquetWriteFn)(nil)).Elem())
+}
+
+// Read reads a set of files and returns lines as a PCollection<elem>
+// based on the internal avro schema of the file.
+// A type - reflect.TypeOf( YourType{} ) -  with
+// JSON tags can be defined or if you wish to return the raw JSON string,
+// use - reflect.TypeOf("") -
+func Read(s beam.Scope, glob string, t reflect.Type) beam.PCollection {
+	s = s.Scope("parquetio.Read")
+	filesystem.ValidateScheme(glob)
+	return read(s, t, beam.Create(s, glob))
+}
+
+func read(s beam.Scope, t reflect.Type, col beam.PCollection) beam.PCollection {
+	files := beam.ParDo(s, expandFn, col)
+	return beam.ParDo(s,
+		&parquetReadFn{Type: beam.EncodedType{T: t}},
+		files,
+		beam.TypeDefinition{Var: beam.XType, T: t},
+	)
+}
+
+func expandFn(ctx context.Context, glob string, emit func(string)) error {
+	if strings.TrimSpace(glob) == "" {
+		return nil // ignore empty string elements here
+	}
+
+	fs, err := filesystem.New(ctx, glob)
+	if err != nil {
+		return err
+	}
+	defer fs.Close()
+
+	files, err := fs.List(ctx, glob)
+	if err != nil {
+		return err
+	}
+	for _, filename := range files {
+		emit(filename)
+	}
+	return nil
+}
+
+type parquetReadFn struct {
+	Type beam.EncodedType
+}
+
+func (a *parquetReadFn) ProcessElement(ctx context.Context, filename string, emit func(beam.X)) (err error) {
```

Comment:
I'd recommend dropping the named return value and explicitly returning the error on each check (you dropped the bare return convention after the first two checks anyway)